adapters/adapters-base/src/main/java/org/apache/cassandra/sidecar/adapters/base/CassandraMetricsOperations.java

```diff
@@ -250,4 +295,186 @@ private List<ClientConnectionEntry> statsToEntries(Stream<ConnectedClientStats>
+    {
+        // Get rates from meter metric for TotalCompactionsCompleted
+        String metricObjectType = String.format(METRICS_OBJ_TYPE_COMPACTION, "TotalCompactionsCompleted");
+        MeterMetricsJmxOperations metricsProxy = jmxClient.proxy(MeterMetricsJmxOperations.class, metricObjectType);
+
+        // Convert per-second rates to the specification:
+        // meanRate: compactions per hour
+        // fifteenMinuteRate: compactions per minute for last 15 minutes
+        double meanRateValue = metricsProxy.getMeanRate() * 3600; // Convert per second to per hour
+        double fifteenMinuteRateValue = metricsProxy.getFifteenMinuteRate() * 60; // Convert per second to per minute
+
+        String meanRate = String.format("%.2f/hour", meanRateValue);
+        String fifteenMinuteRate = String.format("%.2f/minute", fifteenMinuteRateValue);
+
+        return new CompactionStatsResponse.CompletedCompactionsRate(meanRate, fifteenMinuteRate);
+    }
+
+
+    private List<ActiveCompactionEntry> getActiveCompactions(List<Map<String, String>> compactions)
+    {
+        return compactions.stream().map(compactionInfo -> {
+            // Extract fields according to specification
+            String id = compactionInfo.getOrDefault(COMPACTION_ID, DEFAULT_STRING_VALUE);
+            String keyspace = compactionInfo.getOrDefault(KEYSPACE, DEFAULT_STRING_VALUE);
+            String columnFamily = compactionInfo.getOrDefault(COLUMNFAMILY, DEFAULT_STRING_VALUE);
+            String taskType = compactionInfo.getOrDefault(TASK_TYPE, DEFAULT_STRING_VALUE);
+            
+            // Parse byte values
+            long completedBytes = safeParseLong(compactionInfo.getOrDefault(COMPLETED, DEFAULT_NUMBER_VALUE), "completed bytes");
+            long totalBytes = safeParseLong(compactionInfo.getOrDefault(TOTAL, DEFAULT_NUMBER_VALUE), "total bytes");
+            
+            // Calculate percentage completed
+            double percentCompleted = totalBytes == -1 ? 0.0 : (double) completedBytes / totalBytes * 100.0;
+            
+            // Parse SSTables list
+            String ssTablesStr = compactionInfo.getOrDefault(SSTABLES, DEFAULT_STRING_VALUE);
+            List<String> ssTables = ssTablesStr.isEmpty() ? List.of() : List.of(ssTablesStr.split(","));
+            
+            String targetDirectory = compactionInfo.getOrDefault(TARGET_DIRECTORY, DEFAULT_STRING_VALUE);
+            
+            return new ActiveCompactionEntry(id, keyspace, columnFamily, taskType, 
+                                             completedBytes, totalBytes, percentCompleted, 
+                                             ssTables, targetDirectory);
+        }).collect(Collectors.toList());
+    }
+
+    private String calculateRemainingTimeSeconds(final List<ActiveCompactionEntry> activeCompactions,
+                                               final StorageJmxOperations storageService)
```

Comment:
NIT remove final keyword
```suggestion
    private String calculateRemainingTimeSeconds(List<ActiveCompactionEntry> activeCompactions,
                                                StorageJmxOperations storageService)
```

Reply:
Done